packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx

```diff
@@ -63,31 +63,44 @@ export function ProfileTabContent(props: ProfileTabContentProps) {
         currentIdentity.identifier.userId === identity.identifier.userId &&
         personaList.findIndex((persona) => persona?.persona === currentConnectedPersona?.publicHexKey) === -1
 
+    const translate = usePluginI18NField()
     const activatedPlugins = useActivatedPluginsSNSAdaptor('any')
-    const tabs = useAvailablePlugins(activatedPlugins)
-        .flatMap((x) => x.ProfileTabs?.map((y) => ({ ...y, pluginID: x.ID })) ?? [])
-        .filter((z) => z.Utils?.shouldDisplay?.(identity, addressNames) ?? true)
-        .sort((a, z) => {
-            // order those tabs from next id first
-            if (a.pluginID === PluginId.NextID) return -1
-            if (z.pluginID === PluginId.NextID) return 1
-
-            // order those tabs from collectible first
-            if (a.pluginID === PluginId.Collectible) return -1
-            if (z.pluginID === PluginId.Collectible) return 1
-
-            // place those tabs from debugger last
-            if (a.pluginID === PluginId.Debugger) return 1
-            if (z.pluginID === PluginId.Debugger) return -1
-
-            // place those tabs from dao before the last
-            if (a.pluginID === PluginId.DAO) return 1
-            if (z.pluginID === PluginId.DAO) return -1
-
-            return a.priority - z.priority
-        })
-
-    const selectedTabComputed = selectedTab ?? first(tabs)
+    const availablePlugins = useAvailablePlugins(activatedPlugins)
+    const displayPlugins = useMemo(() => {
+        return availablePlugins
+            .flatMap((x) => x.ProfileTabs?.map((y) => ({ ...y, pluginID: x.ID })) ?? [])
+            .filter((z) => z.Utils?.shouldDisplay?.(identity, addressNames) ?? true)
+    }, [availablePlugins, identity, addressNames])
+    const tabs = useMemo(() => {
+        return displayPlugins
+            .sort((a, z) => {
```

Comment:
extract the comparator as function?

```ts
displayPlugins
    .sort(makeComparator(PluginId.NextID, PluginId.Collectible, /* ..., etc */))

function makeComparator(...pluginIds: PluginId[]) {
    return (a, b) => /* comparator logic */
}
```

Reply:
Over context of this PR. I would reject. You are free to refactor in another PR.